File: Java/100-SameTree.java
class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if (p == null || q == null){
            return p == null && q== null; 
        }
        
        if (p.val == q.val){
            return true & isSameTree(p.left, q.left) & isSameTree(p.right, q.right);
        }
        else{
            return false;
        }
    }
}

/*
 Space: O(1)
 Time: O(n), n is the size biggeest tree
*/

File: Python/100-SameTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if (not p) and (not q):
            return True
        elif p and q and p.val == q.val:
            sameL = self.isSameTree(p.left, q.left)
            sameR = self.isSameTree(p.right, q.right)
            
            return sameL and sameR
        else:
            return False
        